modules/mfa/src/Auth/Process/Mfa.php

```diff
@@ -938,6 +939,60 @@ public static function sendManagerCode(array &$state, LoggerInterface $logger):
         $httpUtils->redirectTrustedURL($url, ['mfaId' => $mfaOption['id'], 'StateId' => $stateId]);
     }
 
+    /**
+     * Send a rescue code to the specified recovery contact, then redirect the
+     * user to a page where they can enter the code.
+     *
+     * If successful, this function triggers a redirect and does not return
+     *
+     * @param array $state The state data.
+     * @param string $mfaRecoveryContactID The identifier for the selected MFA
+     *     recovery contact.
+     * @param LoggerInterface $logger A PSR-3 compatible logger.
+     * @throws Exception
+     */
+    public static function sendRecoveryCode(
```

Comment:
The ways this function differs from `sendManagerCode()` can be seen [here](https://github.com/silinternational/ssp-base/pull/322/files/4d8f8fb07c7fc5b61d4b1ba5aff305930ad18da7..66cb356243a533433eeba34105f6a1b2205db100#diff-1c6d19de8c047652a34cef628454dc24b9c7349c439efc4617a7251754b6e712R943).